contracts/test/LightClientUpgrade.t.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+pragma experimental ABIEncoderV2;
+
+import { Test } /*, console2*/ from "forge-std/Test.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { LightClient as LCV1 } from "../src/LightClient.sol";
+import { LightClientV2 as LCV2 } from "../test/LightClientV2.sol";
+import { DeployLightClientContractScript } from "../script/DeployLightClient.s.sol";
+import { UpgradeLightClientScript } from "../script/UpgradeLightClient.s.sol";
+
+contract LightClientUpgradeTest is Test {
+    LCV1 public lcV1Proxy;
+    LCV2 public lcV2Proxy;
+
+    DeployLightClientContractScript public deployer = new DeployLightClientContractScript();
+    UpgradeLightClientScript public upgrader = new UpgradeLightClientScript();
+
+    LCV1.LightClientState public stateV1;
+
+    address public admin;
+    address public proxy;
+
+    // deploy the first implementation with its proxy
+    function setUp() public {
+        (proxy, admin, stateV1) = deployer.run();
+        lcV1Proxy = LCV1(proxy);
+    }
+
+    function testCorrectInitialization() public {
+        assert(lcV1Proxy.blocksPerEpoch() == 10);
+        assert(lcV1Proxy.currentEpoch() == 0);
+
+        assertEq(abi.encode(lcV1Proxy.getGenesisState()), abi.encode(stateV1));
+
+        assertEq(abi.encode(lcV1Proxy.getFinalizedState()), abi.encode(stateV1));
+
+        bytes32 stakeTableComm = lcV1Proxy.computeStakeTableComm(stateV1);
+        assertEq(lcV1Proxy.votingStakeTableCommitment(), stakeTableComm);
+        assertEq(lcV1Proxy.frozenStakeTableCommitment(), stakeTableComm);
+        assertEq(lcV1Proxy.votingThreshold(), stateV1.threshold);
+        assertEq(lcV1Proxy.frozenThreshold(), stateV1.threshold);
+    }
+
+    // that the data remains the same after upgrading the implementation
+    function testUpgradeSameData() public {
+        // Upgrade LightClient and check that the genesis state is not changed and that the new
+        // field
+        // of the upgraded contract is set to 0
+        lcV2Proxy = LCV2(upgrader.run(admin, proxy));
+
+        assertEq(lcV2Proxy.newField(), 0);
+        assertEq(lcV2Proxy.blocksPerEpoch(), 10);
+        assertEq(lcV2Proxy.currentEpoch(), 0);
+
```

Comment:
I think it would be worthwhile to assert the extra field added to the LightClientState struct is also equal to what we expect (0) - `assertEq(lcV2Proxy.getFinalizedState().extraField, 0);`

Reply:
As discussed, this equality is already checked in https://github.com/EspressoSystems/espresso-sequencer/blob/96a03a0ba34fbaf080bc3020ab21a4b8258ac0d0/contracts/test/LightClientUpgrade.t.sol#L65